### PR TITLE
FEATURE: Add support for Category Description Translations

### PR DESCRIPTION
**Pull request by merefield**

### CODE REVIEW

File: assets/javascripts/discourse/lib/multilingual-category.js.es6 [hunk 1]
```diff
@@ -16,6 +16,15 @@ function translatedCategoryName(category) {
   return translatedName || get(category, "name");
 }
 
+function translatedCategoryDescription(category) {
```
Comment (angusmcleod) [inline assets/javascripts/discourse/lib/multilingual-category.js.es6:19]: @merefield I feel like this logic could be simplified. It feels a bit spaghetti.

File: plugin.rb [hunk 2]
```diff
@@ -201,8 +201,14 @@
     Multilingual::Translation.get("category_name", object.slug_path)
   end
 
+  add_to_serializer(:basic_category, :description_translations) do
```
Comment (angusmcleod) [inline plugin.rb:204]: @merefield We should add a spec for the description translation themselves. I think we're just testing the upload atm? Perhaps in the next PR.

File: assets/javascripts/discourse/lib/multilingual-category.js.es6 (hunk 1 above)
Comment (merefield) [reply assets/javascripts/discourse/lib/multilingual-category.js.es6:19]: hmmm .... the challenge is this:
```
function translatedCategoryDescription(category) {
  let translatedDescription;
  let descriptionTranslations = get(category, "description_translations");
  if (descriptionTranslations) {
    translatedDescription = descriptionTranslations[I18n.currentLocale()];
  }
  return translatedDescription || get(category, "description_excerpt");
}
```
if the attribute doesn't exist, current code protects you
you could do this:
```
function translatedCategoryDescription(category) {
  return get(category, "description_translations")[I18n.currentLocale()] ||
    get(category, "description_excerpt");
}
```
but you risk an error if the attribute is missing.

Comment (merefield) [reply assets/javascripts/discourse/lib/multilingual-category.js.es6:19]: As discussed, this is to protect against the attribute being missing.

File: plugin.rb (hunk 2 above)
Comment (merefield) [reply plugin.rb:204]: Moved to a different spec file to cover translation scope specifically.

**Pull request merged**